### PR TITLE
RS/YJ/RULE 21-13

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
I'd appreciate it if you could review this PR. Thank you!

### CODE REVIEW

File: rct229/rules/section21/section21rule13.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        fluid_loop_b = context_item.baseline
+        return getattr_(fluid_loop_b, "FluidLoop", "type") == FLUID_LOOP.HEATING
+
+    class HeatingFluidLoopRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section21Rule13.HeatingFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["heating_design_and_control"],
+                    "$.minimum_flow_fraction ": ["minimum_flow_fraction"],
```
Comment (weilixu) [inline rct229/rules/section21/section21rule13.py:72]: This should be `"$.heating_design_and_control" : ["minimum_flow_fraction"]`

File: rct229/rules/section21/section21rule13.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        def get_calc_vals(self, context, data=None):
+            heating_fluid_loop_b = context.baseline
+            minimum_flow_fraction = heating_fluid_loop_b["heating_design_and_control"][
+                "minimum_flow_fraction"
+            ]
+            return {"minimum_flow_fraction": minimum_flow_fraction}
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            return True if calc_vals["minimum_flow_fraction"] == 0.25 else False
```
Comment (weilixu) [inline rct229/rules/section21/section21rule13.py:84]: can be 
```python
return calc_vals["minimum_flow_fraction"] == 0.25
```

For style, I prefer to have a global Constant:
```
MINIMUM_TURNDOWN_RATIO = 0.25
```
and then in the logic use:
```python
return calc_vals["minimum_flow_fraction"] == MINIMUM_TURNDOWN_RATIO
```

**Review by weilixu — APPROVED** [review]
Approved with one issue must be addressed and a style comment

File: rct229/rules/section21/section21rule13.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rules/section21/section21rule13.py]: Fixed.

File: rct229/rules/section21/section21rule13.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rules/section21/section21rule13.py]: Fixed.

File: rct229/rules/section21/section21rule13.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        fluid_loop_b = context_item.baseline
+        return getattr_(fluid_loop_b, "FluidLoop", "type") == FLUID_LOOP.HEATING
+
+    class HeatingFluidLoopRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section21Rule13.HeatingFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["heating_design_and_control"],
+                    "$.heating_design_and_control": ["minimum_flow_fraction"],
```
Comment (charliepnnl) [inline rct229/rules/section21/section21rule13.py:74]: The "$." on line 74 is not needed.

File: rct229/rules/section21/section21rule13.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+        def get_calc_vals(self, context, data=None):
+            heating_fluid_loop_b = context.baseline
+            minimum_flow_fraction = heating_fluid_loop_b["heating_design_and_control"][
+                "minimum_flow_fraction"
+            ]
+            return {"minimum_flow_fraction": minimum_flow_fraction}
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            return (
```
Comment (charliepnnl) [inline rct229/rules/section21/section21rule13.py:86]: This logic in the return statement can be simplified. The function body could be:
```
minimum_flow_fraction = calc_vals["minimum_flow_fraction"]

return minimum_flow_fraction == MINIMUM_TURNDOWN_RATIO
```

**Review by charliepnnl — APPROVED** [review]
Approved with just a couple of stylistic comments

**Comment by yunjoonjung-PNNL** [comment]
I addressed all of Charlie's comments and would appreciate it if you could check the new updates. Thank you!

**Comment by weilixu** [comment]
Looks good.

**Pull request merged**